metadata-ingestion/src/datahub/ingestion/source/sql/mssql/job_models.py

```diff
@@ -0,0 +1,240 @@
+
+
+@dataclass
+class MSSQLProceduresContainer:
+    db: str
+    platform_instance: str
+    name: str
+    env: str
+    source: str = "mssql"
+    type: str = "JOB"
+
+    @property
+    def formatted_name(self) -> str:
+        return f"{self.formatted_platform_instance}.{self.name.replace(',', '-')}"
+
+    @property
+    def orchestrator(self) -> str:
+        return self.source
+
+    @property
+    def formatted_platform_instance(self) -> str:
+        return self.platform_instance.replace(".", "/")
+
+    @property
+    def cluster(self) -> str:
+        return f"{self.env}"
+
+    @property
+    def full_type(self) -> str:
+        return f"({self.source},{self.name},{self.env})"
+
+
+@dataclass
+class ProcedureParameter:
+    name: str
+    type: str
+
+    @property
+    def properties(self):
+        return {"type": self.type}
+
+
+@dataclass
+class StoredProcedure:
+    db: str
+    schema: str
+    name: str
+    flow: Union[MSSQLJob, MSSQLProceduresContainer]
+    type: str = "STORED_PROCEDURE"
+    source: str = "mssql"
+
+    @property
+    def full_type(self) -> str:
+        return self.source.upper() + "_" + self.type
+
+    @property
+    def formatted_name(self) -> str:
+        return self.name.replace(",", "-")
+
+    @property
+    def full_name(self) -> str:
+        return f"{self.db}.{self.schema}.{self.formatted_name}"
+
+    @property
+    def escape_full_name(self) -> str:
+        return f"[{self.db}].[{self.schema}].[{self.formatted_name}]"
+
+
+@dataclass
+class JobStep:
+    job_name: str
+    step_name: str
+    flow: MSSQLJob
+    type: str = "JOB_STEP"
+    source: str = "mssql"
+
+    @property
+    def formatted_step(self) -> str:
+        return self.step_name.replace(",", "-").replace(" ", "_").lower()
+
+    @property
+    def formatted_name(self) -> str:
+        return self.job_name.replace(",", "-")
+
+    @property
+    def full_type(self) -> str:
+        return self.source.upper() + "_" + self.type
+
+    @property
+    def full_name(self) -> str:
+        return f"{self.formatted_name}.{self.formatted_name}"
+
+
+@dataclass
+class MSSQLDataJob:
+    entity: Union[StoredProcedure, JobStep]
+    type: str = "dataJob"
+    source: str = "mssql"
+    external_url: str = ""
+    description: Optional[str] = None
+    status: Optional[str] = None
+    incoming: List[str] = field(default_factory=list)
+    outgoing: List[str] = field(default_factory=list)
+    input_jobs: List[str] = field(default_factory=list)
+    job_properties: Dict[str, str] = field(default_factory=dict)
+
+    @property
+    def urn(self):
+        return make_data_job_urn(
+            orchestrator=self.entity.flow.orchestrator,
+            flow_id=self.entity.flow.formatted_name,
+            job_id=self.entity.formatted_name,
+            cluster=self.entity.flow.cluster,
+        )
+
+    def add_property(self, name, value):
+        self.job_properties[name] = value
+
+    @property
+    def valued_properties(self):
```

Comment:
can we add type annotations to this file?

Reply:
done